### DISCUSSION TITLE
How to get the user profile id in the bot

### DISCUSSION THREAD

**Discussion by okaeiz** [kianahmadian/python-bale-bot#19] [Q&A] [answered]
Currently, when a message is sent by the users, I can get their id using `message.author` attribute. But it is not their real id which is visible in their profiles. Instead, it returns a longer string. How can I get their real id?

**Comment by kian-ahmadian** [comment, marked as answer]
**Please read the document first and then open an issue here about your question.** The `message.author` attribute is the same as the `bale.User` object, just like the explanation I gave earlier.  Now you can see inside the document, for example, what attributes `bale.User` has that you can use to get the username or numeric ID and anything else related to it.

[Document](https://python-bale-bot.ir/docs)
